great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -370,6 +370,9 @@ def get_config_with_variables_substituted(
             )
         )
 
+    def delete_datasource(self, datasource_name: str) -> None:  # type: ignore[override]
```

Comment:
Can we update the parent class signature so that we aren't doing an LSP violation here?

Reply:
Is `datasource_name` being `Optional` on the Superclass correct?

I'm guessing `save_changes` isn't an option because it's always going to be true? To deal with that we can raise an error if they provide something other than false. And that way we preserve LSP.

```python
def delete_datasource(self, datasource_name: str, save_changes: bool = True) -> None:
    if save_changes is not True:
        raise ValueError(f"`save_changes` for {self.__class__.__name__}.delete_datasource must be `True`")
    ...
```

Reply:
I think we want to remove `save_changes` entirely but that will be a more involved effort (probably immediately following this PR)

Reply:
The only remaining issue I have here is with point `2`, that LSP violation is likely to actually break existing setups for users if they swap their Datacontext class to a Cloud one.

Example... currently this would break, if they swapped the Datasource type to cloud.
```python
my_datacontext.delete_dastsource("datasourous_rex", save_changes=True)
```

But really the only reason this should break is if they tried to do this...
```python
my_datacontext.delete_dastsource("datasourous_rex", save_changes=False)
```

But we can entirely remove this pain point for moving to cloud with the suggested change above.